pkg/asset/machines/baremetal/machines.go

```diff
@@ -60,11 +62,18 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
-func provider(clusterName string, networkInterfaceAddress string, platform *baremetal.Platform, userDataSecret string) *baremetalprovider.BareMetalMachineProviderSpec {
+func provider(clusterName string, networkInterfaceAddress string, platform *baremetal.Platform, osImage string, userDataSecret string) *baremetalprovider.BareMetalMachineProviderSpec {
+	// The rhcos-downloader container launched by the baremetal-operator downloads the image,
+	// compresses it to speed up deployments and makes it available on provisioningIP via http
+	provisioningIP := platform.ClusterProvisioningIP
+	imageFilename := path.Base(osImage)
+	compressedImageFilename := strings.Replace(imageFilename, "openstack", "compressed", 1)
+	cacheImageURL := fmt.Sprintf("http://%s:6180/images/%s/%s", provisioningIP, imageFilename, compressedImageFilename)
+	cacheChecksumURL := fmt.Sprintf("%s.md5sum", cacheImageURL)
 	return &baremetalprovider.BareMetalMachineProviderSpec{
 		Image: baremetalprovider.Image{
```

Comment:
how does it know what version of image to load? the URL seems like a URL to a directory?

Reply:
I might be wrong, therefore would love to see a comment here...

Reply:
Same as above, it's aligning with the file paths/URL generated via rhcos-downloader, so I'll add a comment that clarifies both the filename and the URL.